### ISSUE TITLE
Impossible to enter into MODE

### ISSUE THREAD

**Issue by StanMOI** [mitchellcairns/gbc-rgb-led-2.0#2]
Hi,
I trying to enter into MODE's but nothing happened.
Why?
On my GBC I have an IPS screen that also using mode by holding "start" and "select" buttons.
Do you know my IPS screen can be the reason to my problem ?
Maybe I should try to use some other button vs "select", just to see if its make working ?
Thanks for help.

**Comment by mitchellcairns** [comment]
I believe it's some conflict with the external device (IPS board) pulling up at a different resistance value.

**Comment by StanMOI** [comment]
> I believe it's some conflict with the external device (IPS board) pulling up at a different resistance value.

I made test circuit with your code , Attiny85 DIP-8 and push buttons.
It's doesn't work's.
LED's bright  but impossible to enter into MODE options.
![IMG_4713](https://github.com/HandHeldLegend/gbc-rgb-led-2.0/assets/38383423/3d20e29c-656f-4673-bacb-499084f7bdb0)

**Comment by mitchellcairns** [comment]
You'll have to forgive me, it's been a very very long time since I've worked on this project :(

**Comment by StanMOI** [comment]
> You'll have to forgive me, it's been a very very long time since I've worked on this project :(

Most be some code error line.
Hope you can help me, please.

**Comment by StanMOI** [comment]
Here is schematic 
<img width="1799" alt="Capture d’écran 2023-11-03 à 00 48 30" src="https://github.com/HandHeldLegend/gbc-rgb-led-2.0/assets/38383423/eebc6b9c-8e82-4c1b-9581-c883e555c2b3">

**Comment by StanMOI** [comment]
Even when I add into circuit the "pull-up" resistors (10K) for buttons, it's still not working.
<img width="722" alt="Capture d’écran 2023-11-03 à 12 28 54" src="https://github.com/HandHeldLegend/gbc-rgb-led-2.0/assets/38383423/16a72256-5d32-4d3e-bf76-93aa9e1460d5">

**Comment by StanMOI** [comment]
My precedent draw for 10K resistors was wrong.
I corrected my circuit with "PullUp" resistors connected to 5V. (and not to Attiny directly)
But nothing changes.
Only White light brighting. Like before, Impossible to enter into "MODE" function.

<img width="854" alt="Capture d’écran 2023-11-10 à 09 40 54" src="https://github.com/HandHeldLegend/gbc-rgb-led-2.0/assets/38383423/5de0064e-bffc-43b9-a315-f956628307a6">
<img width="1599" alt="Capture d’écran 2023-11-10 à 09 41 43" src="https://github.com/HandHeldLegend/gbc-rgb-led-2.0/assets/38383423/f6c58452-f991-4ec3-813d-7dd8ed594e4d">

**Comment by mitchellcairns** [comment]
Did you properly flash the fuses on the ATTINY85 to run at the appropriate clock speed for FastLED. I do not have instructions or offer support on that part sorry

**Comment by StanMOI** [comment]
> Did you properly flash the fuses on the ATTINY85 to run at the appropriate clock speed for FastLED. I do not have instructions or offer support on that part sorry

I flashed Attiny with your « .HEX » file and « TL866 » programmer.
I see that it was correctly flashed by all lighting LED's .
If I remplace flashed Attiny by an empty Attiny then no one LED lighting.
I also successfully flashed the same Attiny with Arduino.
In your Arduino code I changed the total quantity of LED's, just to see if your code work's, and YES it takes in consideration.
So flashing works with both.

One interesting thing...
Using same schematic, I flashed Attiny with one exemple code call's "Blink" (Arduino library "FastLED").
Here also it takes in consideration the total quantity of LED's (if I change it), but in this exemple I see no "Blink" effect.
I check twice my connection of each WS2812 LED, all good like on pinout picture. Circuit Powered with 5v.
Also made one test with LED strip WS2812 = same, no blink effect.
Then I put off all resistors and buttons, and plug into this circuit only LED strip = same result, no blink effect. All leds only bright with white color.
Strange, strange...

![WS2812](https://github.com/HandHeldLegend/gbc-rgb-led-2.0/assets/38383423/c11581d4-6cf0-4a4e-a9f7-53a2467434bc)

**Comment by mitchellcairns** [comment]
Flashing the firmware is separate from flashing the appropriate fuse values.

The default color scheme is Red orange yellow green blue purple, if it's all white something else is wrong.

**Comment by StanMOI** [comment]
> Flashing the firmware is separate from flashing the appropriate fuse values.
>
What does it mean?

I done one test with Arduino UNO vs Attiny85 using "FastLED" exemple code "Blink".
The result is that works with UNO. And not with Attiny85. 
:(

Here is my setup for flashing Attiny85 with UNO:
Before all, I done "Burn Bootloader" to my UNO.
<img width="263" alt="Capture d’écran 2023-11-11 à 20 14 26" src="https://github.com/HandHeldLegend/gbc-rgb-led-2.0/assets/38383423/186b8857-9d20-4087-ab1a-f76782734990">

<img width="246" alt="Capture d’écran 2023-11-11 à 20 28 14" src="https://github.com/HandHeldLegend/gbc-rgb-led-2.0/assets/38383423/7010fbe5-be3c-4a88-8203-774945b22989">

**Comment by mitchellcairns** [comment]
I'd try burn bootloader, then upload again, cheers

**Comment by StanMOI** [comment]
> I'd try burn bootloader, then upload again, cheers
>
Bootloader done.
When I just click to "upload" I get error message "programmer is required to upload".
So instead of clicking the "Upload" to uploading to my ATTiny85 I do:
Sketch > Upload Using Programmer (from the Arduino IDE menus).

**Comment by StanMOI** [comment]
Mitch I got it!

The mistake was "electrolytic capacitor" that I keep it onto my Arduino UNO board.
This capacitor shouldn't be here when you need to "prepare" the Arduino for the fist time by uploading the ISP sketch to it.

![attiny_programming_w_arduino_uno 2](https://github.com/HandHeldLegend/gbc-rgb-led-2.0/assets/38383423/e4be5df6-0e0f-4564-9140-4801a314d2c8)
So, the reason of this "error" was my Arduino UNO that wasn't be prepared correctly before.

So here is how to do exactly :

**1- Before all = > Prepare your Arduino UNO to be able to flash Attiny85.**
Attention, Arduino UNO "reset" pin have to be without any capacitor.
- Go and select: Tools/Board/Arduino AVR boars/Arduino UNO
- Go and select: Tools/Board/ Programmer/Arduino as ISP
<img width="277" alt="Capture d’écran 2023-11-11 à 22 30 50" src="https://github.com/HandHeldLegend/gbc-rgb-led-2.0/assets/38383423/d4cfc28d-4159-483f-ac3d-d35ca8730bd3">

- Go and select: File/Exemples/11.ArduinolSP/ArduinoISP
<img width="567" alt="Capture d’écran 2023-11-11 à 22 32 13" src="https://github.com/HandHeldLegend/gbc-rgb-led-2.0/assets/38383423/7b7b78eb-9ef1-4b24-9a3c-646d1a42a3f1">

- Upload ArduinoISP Exemple Code by clicking to "upload" button
<img width="531" alt="Capture d’écran 2023-11-11 à 22 34 19" src="https://github.com/HandHeldLegend/gbc-rgb-led-2.0/assets/38383423/4f4200ab-ed45-49db-bb12-239a81e04094">

Done! Arduino are almost ready.

Now you can configure your Arduino UNO pinout for flashing Attiny85.
![attiny_programming_w_arduino_uno](https://github.com/HandHeldLegend/gbc-rgb-led-2.0/assets/38383423/c4d2beaa-150a-444e-8554-e10985976908)

**2 - Install Attiny boards manager by David A. Mellis**
- Go to : Arduino IDE/Preferences/
- Add URL to the line "Additional boards manager URLs":
https://raw.githubusercontent.com/digistump/arduino-boards-index/master/package_digistump_index.json
<img width="780" alt="Capture d’écran 2023-11-11 à 22 36 51" src="https://github.com/HandHeldLegend/gbc-rgb-led-2.0/assets/38383423/ca3b34f7-2d30-4663-ad43-cf889d9e2602">


- Go to : Tools/Board/Boards Manager/
- in search line type "attiny" and find "attiny by David A. Mellisby"


<img width="288" alt="Capture d’écran 2023-11-11 à 22 35 48" src="https://github.com/HandHeldLegend/gbc-rgb-led-2.0/assets/38383423/3b24236b-d571-48db-b704-f030b032ccdb">

- Install it

**3-Prepare Arduino UNO to flash Attiny85:**

- Go and select : Tools/Board/attiny/ATtiny25/45/85
- Go and select : Tools/Clock: "Internal 8 MHz"
- Go and select : Tools/Processor: "ATtiny85"
- Go and select : Tools/Programmer: "Arduino as ISP"
<img width="615" alt="Capture d’écran 2023-11-11 à 22 38 56" src="https://github.com/HandHeldLegend/gbc-rgb-led-2.0/assets/38383423/411db277-4aef-4aca-a7af-0642fc9f6fec">

- Go to : Tools/Burn Bootloader
<img width="278" alt="Capture d’écran 2023-11-11 à 22 42 03" src="https://github.com/HandHeldLegend/gbc-rgb-led-2.0/assets/38383423/31b0e7ce-2a26-4a1e-966f-5b42d43df0ee">

**4-Flash sketch into Attiny85:**

- Open sketch if it not done before.
- Go to "Sketch" and select "Upload Using Programmer"
<img width="252" alt="Capture d’écran 2023-11-11 à 22 42 55" src="https://github.com/HandHeldLegend/gbc-rgb-led-2.0/assets/38383423/3e9746a7-a6a0-454c-8bcb-9b504d528f64">

DONE !

**Comment by mitchellcairns** [comment]
A+ great investigation and happy coding :)

**Comment by StanMOI** [comment]
> A+ great investigation and happy coding :)

Haha...
Sorry, but not so fast Mitch.
An other problem are here.

When I unplug Arduino UNO from my testing board and plug it wit 5V of external power,
it's doesn't work any more! 
Only leds brights and no reaction when I touching two buttons to enter into MODE functions.

As in your code you was using "INPUT_PULLUP" so I added 10K "pull-up resistor"s to each button.
You can see it on my schematic. 
It's look that "pull-up resistors" not helps when Arduino are unplugged .
I know that Arduino has his own "pull-up resistor". Maybe that's why it's working with Arduino and not without?
Then, always in the same configuration (5V external power), I tried to invert resistors to "pull-down" position, and see if it work's. YES it works !?

At the same time I ordered your ready flex pcb from [](https://handheldlegend.com)
[](https://handheldlegend.com/products/game-boy-color-rgb-led-button-kit)
I soldered it to my GBC and unfortunately it's also not working.
Same thing = No reaction when you press two buttons to enter into MODE, only leds lighting.
All soldered points are good, I tested it with a multimeter and "peep" function.

Some says that your flex board are not compatible with some IPS screens.
So I unplugged Screen IPS cable from mother board to see if it work's = NO CHANCE
Then I desoldered your flex boar and powered it with an external 5V, and 3 conductive buttons (to imitate GBC buttons) = still NO REACTION.

I'am disappeared  :(

**Comment by mitchellcairns** [comment]
If you have a capacitor on the voltage line to the microcontroller you need a way for that power to drain, otherwise you can get brownout condition

**Comment by StanMOI** [comment]
> If you have a capacitor on the voltage line to the microcontroller you need a way for that power to drain, otherwise you can get brownout condition

Exact Mitch !
![280148447-3d20e29c-656f-4673-bacb-499084f7bdb0](https://github.com/HandHeldLegend/gbc-rgb-led-2.0/assets/38383423/c90a3d18-3e0a-4f91-b639-f768cfb0d4b8)
I removed this electrolytic capacitor from my testing board and it's work again.
Alleluia ! 

What about "pull-up resistors" ?
I tried to remove them from my testing board = it's work's
On your flex board you don't have it = it's work's too.
I tried to add them onto your flex = it's work's too.
Do GBC already have those resistors on motherboard, so that's why you don't added them on your flex design?

About IPS screen and your flex board I found solution.
As my IPS screen also has functions when pushing "select" and "start" buttons, I just tied those wires to "A" and "B" buttons on your flex board.
Now your flex work's with my IPS screen. IPS screen function working fine (by pressing A & B buttons).
My testing board working too !

Thank you for support Mitch.
